pystac/extensions/scientific.py

```diff
@@ -0,0 +1,272 @@
+
+        doi_str = parse.unquote(urlsplit.path[1:])  # Drop leading '/'
+        if not is_doi_valid(doi_str):
+            raise pystac.STACError('Not valid DOI or DOI URL: "%s"' % doi)
+
+        self.doi = doi_str
+
+    def __repr__(self) -> str:
+        return f'<Doi {self.doi}>'
+
+    def __eq__(self, other: _DoiType) -> bool:
+        """Compares with the other using case in-sensitive check on self.doi."""
+        if not isinstance(other, Doi):
+            return False
+
+        return self.doi.lower() == other.doi.lower()
+
+    def url(self) -> str:
+        return DOI_URL_BASE + parse.quote(self.doi)
+
+
+class Publication:
+    """Helper for Publication entries."""
+    def __init__(self, doi: Union[str, Doi], citation: str) -> None:
+        if not isinstance(doi, Doi):
+            doi = Doi(doi)
+        self.doi = doi
+        self.citation = citation
+
+    def __eq__(self, other: _PublicationType) -> bool:
+        if not isinstance(other, Publication):
+            return NotImplemented
+
+        return self.doi == other.doi and self.citation == other.citation
+
+    def __repr__(self) -> str:
+        return f'<Publication doi={self.doi} target={self.citation}>'
+
+    def to_dict(self) -> Dict[str, str]:
+        return copy.deepcopy({'doi': self.doi.doi, 'citation': self.citation})
+
+    @staticmethod
+    def from_dict(d: Dict[str, str]) -> _PublicationType:
+        return Publication(d['doi'], d['citation'])
+
+    def get_link(self) -> link.Link:
+        url = self.doi.url()
+        return link.Link(CITE_AS, url)
+
+
+def remove_link(links: List[link.Link], pub: Publication):
+    url = pub.doi.url()
+    for i, a_link in enumerate(links):
+        if a_link.rel != CITE_AS:
+            continue
+        if a_link.target == url:
+            del links[i]
+            break
+
+
+class ScientificItemExt(base.ItemExtension):
+    """Add an citation and dois to a STAC Item."""
+    def __init__(self, an_item: item.Item) -> None:
+        self.item = an_item
+
+    def apply(self,
+              doi: Optional[Union[str, Doi]] = None,
+              citation: Optional[str] = None,
+              publications: Optional[List[Publication]] = None) -> None:
+        if doi:
+            self.doi = doi
+        if citation:
+            self.citation = citation
+        if publications:
+            self.publications = publications
+
+    @classmethod
+    def from_item(cls: _ScientificItemExtType, an_item: item.Item) -> _ScientificItemExtType:
+        return cls(an_item)
+
+    @classmethod
+    def _object_links(cls: _ScientificItemExtType) -> List:
+        return []
+
+    @property
+    def doi(self) -> Doi:
+        return self.item.properties.get(DOI)
+
+    @doi.setter
+    def doi(self, v: Union[str, Doi]) -> None:
+        if isinstance(v, Doi):
+            v = v.doi
+        self.item.properties[DOI] = v
+        url = DOI_URL_BASE + self.doi
+        self.item.add_link(pystac.link.Link(CITE_AS, url))
```

Comment:
This should also clear previously added links.

Reply:
I don't know if we can safely remove all the old publication links.  Case that have me worried:

* What if the doi link and one of the publications are the same?
* What if the user adds a `cite-as` link on their own?
* What if a link gets mutated by some other call?

Reply:
Maybe I don't worry and I just search and destroy based on cite-as links with the doi substring

Reply:
Fixed in https://github.com/stac-utils/pystac/pull/199/commits/af38dc1373818f5a7885ec6d36fd7aee19a80431